sunpy/net/hek/utils.py

```diff
@@ -0,0 +1,175 @@
+            break
+    return table
+
+# NOTE: Needs unit test
+def parse_chaincode(value, idx, attribute, unit_prop):
+    coord1_unit = u.deg
+    coord2_unit = u.deg
+    if attribute["frame"] == "helioprojective":
+        coord1_unit = u.arcsec
+        coord2_unit = u.arcsec
+    elif attribute["frame"] == "heliocentric":
+        coord1_unit = u.R_sun
+        coord2_unit = u.deg
+    elif attribute["frame"] == "icrs":
+        coord1_unit = get_unit("coord1_unit", unit_prop)
+        coord2_unit = get_unit("coord2_unit", unit_prop)
+
+    coordinates_str = value.split('((')[1].split('))')[0]
+    coord1_list = [float(coord.split()[0]) for coord in coordinates_str.split(',')] * coord1_unit
+    coord2_list = [float(coord.split()[1]) for coord in coordinates_str.split(',')] * coord2_unit
+    vertices = {}
+    if attribute["frame"] == "heliocentric":
+       vertices = SkyCoord(coord1_list, coord2_list, [1]* len(coord1_list)* u.AU, representation_type = "cylindrical", frame = "heliocentric")
+    else:
+        vertices = SkyCoord(coord1_list, coord2_list, frame=attribute["frame"])
+
+    return PolygonSkyRegion(vertices = vertices)
+
+# NOTE: Needs unit test
+def get_unit(unit_prop, unit):
+    """
+    Converts string into astropy unit.
+
+    Parameters
+    ----------
+    unit_prop: str
+        The unit parameter from HEK api.
+    unit: str
+        The targeted unit
+
+    Returns
+    -------
+        unit
+            Astropy unit object (e.g. <class 'astropy.units.core.Unit'> or <class 'astropy.units.core.CompositeUnit'>)
+
+    Raises
+    ------
+    ValueError
+        Because `unit` did not parse as unit.
+
+    Notes
+    ----
+    For the complete list of HEK parameters: https://www.lmsal.com/hek/VOEvent_Spec.html
+
+    """
+    cm2 = u.def_unit("cm2", u.cm**3)
+    m2 = u.def_unit("m2", u.m**2)
+    m3 = u.def_unit("m3", u.m**3)
+
+    aliases = {
+        "steradian": u.sr,
+        "arcseconds": u.arcsec,
+        "degrees": u.deg,
+        "sec": u.s,
+        "Emx": u.Mx,
+        "Amperes": u.A,
+        "ergs": u.erg,
+        "cubic centimeter": u.ml,
+        "square centimeter": cm2,
+        "cubic meter": m3,
+        "square meter": m2,
+    }
+
+    with u.add_enabled_units([cm2, m2, m3]), u.set_enabled_aliases(aliases):
+        if unit_prop in ["coord1_unit", "coord2_unit", "coord3_unit", "event_coordunit"]:
+            coord1_unit, coord2_unit, coord3_unit = None, None, None
+            coord_units = re.split(r'[, ]', unit)
+            if len(coord_units) == 1: # deg
+               coord1_unit = coord2_unit = u.Unit(coord_units[0])
+            elif len(coord_units) == 2:
+                coord1_unit = u.Unit(coord_units[0])
+                coord2_unit = u.Unit(coord_units[1])
+            else:
+                coord1_unit = u.Unit(coord_units[0])
+                coord2_unit = u.Unit(coord_units[1])
+                coord3_unit = u.Unit(coord_units[2])
+            return locals()[unit_prop]
```

Comment:
Do we need to do this?!

Reply:
I think the three units should be the same, so we can just do this:
```python
if unit_prop in ["coord1_unit", "coord2_unit", "coord3_unit", "event_coordunit"]:
    coord_units = re.split(r'[, ]', unit)
    return u.Unit(coord_units[0])

```

Reply:
Can you try that and see how it goes?